src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationKeyVaultOptions.cs

```diff
@@ -15,7 +15,12 @@ namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
     public class AzureAppConfigurationKeyVaultOptions
     {
         internal TokenCredential Credential;
-        internal SecretClientOptions ClientOptions = new SecretClientOptions();
+        internal SecretClientOptions ClientOptions = new SecretClientOptions
+        {
+            Retry = {
+                MaxRetries = int.MaxValue
```

Comment:
How do we enforce an upper limit for the time spent on KeyVault retries?

Reply:
Yea fair question, I'm not sure how to choose a time if our goal with this change is to prevent a refresh from completing without updating a secret that should've been updated if it weren't for the requests failing temporarily.

Reply:
However, I do think the default retry count of 3 is a bit too low. The SDK uses an exponential backoff by strategy, so even a minor bump makes a notable difference.

I did some testing in a console app by inducing a socket exception.
A retry count of 3 was exhausted in 6 seconds.
A retry count of 6 was exhausted in 46 seconds
A retry count of 7 was exhausted in 99 seconds.

I think we could benefit by bumping this up, and I would suggest 6. The reason being that it still falls under our startuptimeout, which could be important for diagnosing an issue like if you reference a key vault domain that doesn't actually exist, given this is classified as retriable.

Reply:
It could be 46 seconds per KVR, is that a concern?

Reply:
> It could be 46 seconds per KVR, is that a concern?

I don't see it as a concern.

Reply:
Discussed with Jimmy. If the first KVR fails to resolve, we throw KeyVaultReferenceException and halt the startup. So 46 seconds is not a concern since its under the allowed startup timeout.